pandas/core/groupby/groupby.py

```diff
@@ -1120,12 +1117,14 @@ def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
             if self.obj.ndim == 2:
                 # i.e. DataFrameGroupBy
                 numeric_only = True
+                obj = self._obj_with_exclusions
+                check = obj._get_numeric_data()
+                if len(obj.columns) and not len(check.columns) and not obj.empty:
+                    numeric_only = False
+
             else:
                 numeric_only = False
-
-        # error: Incompatible return value type (got "Union[bool, NoDefault]",
-        # expected "bool")
-        return numeric_only  # type: ignore[return-value]
```

Comment:
can you restore this too. The mypy error needs to be ignored as mypy does not know that `lib.no_default` is a singleton.

Reply:
> @kurchi1205 can you add a release note.

I assume the release note should be added in v1.4.0.rst

Reply:
> can you restore this too. The mypy error needs to be ignored as mypy does not know that lib.no_default is a singleton.
I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

Reply:
> > @kurchi1205 can you add a release note.
> 
> I assume the release note should be added in v1.4.0.rst

We discussed at pandas dev meeting today, and put this in v1.3.3.rst